templates/cluster-template-crs-cni.yaml

```diff
@@ -47,14 +47,15 @@ spec:
         net.bridge.bridge-nf-call-ip6tables = 1
         EOF
       - sysctl --system
-      - apt-get -y update
-      - DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl
-      - curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-      - echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list
-      - apt-get update -y
-      - TRIMMED_KUBERNETES_VERSION=$(echo {{ .kubernetesVersion }} | sed 's/\./\\./g' | sed 's/^v//')
-      - RESOLVED_KUBERNETES_VERSION=$(apt-cache policy kubelet | awk -v VERSION=$${TRIMMED_KUBERNETES_VERSION} '$1~ VERSION { print $1 }' | head -n1)
-      - apt-get install -y ca-certificates socat jq ebtables apt-transport-https cloud-utils prips containerd kubelet=$${RESOLVED_KUBERNETES_VERSION} kubeadm=$${RESOLVED_KUBERNETES_VERSION} kubectl=$${RESOLVED_KUBERNETES_VERSION}
+      - |
```

Comment:
This change is needed because `clusterctl generate cluster` will otherwise split some of the lines creating incorrect YAML which can cause some of the commands to not be executed.

Reply:
So I'm confused on this one, what about your changes make this necessary when it wasn't needed before?

Reply:
@cprivite It actually is needed now as well, but it doesn't seem to have been causing major issues which is why it might not have been detected.

Without this change, the `preKubeadmCommands` YAML looks like this (using clusterctl 1.1.2 and 1.1.3):

![image](https://user-images.githubusercontent.com/28541758/157474415-5ae7fa78-33d3-4e46-a1e4-baa57aa17a2a.png)

There are errors in the YAML formatting which causes some commands to not be executed.
- The `-` on line 64 should be part of the previous line
- Due to line 68 all following lines are seen as part of the same command
- If you manually fix line 68, line 70 will cause the same issue 

With this change, the formatting issues are avoided as you can see here:

![image](https://user-images.githubusercontent.com/28541758/157475507-e078b0e5-4295-4009-953c-5068fe9ba2e6.png)

I came across this issue while I was trying to figure out why after a certain change to the commands the `kube-vip` command appeared to not have executed properly.

Reply:
Gotcha, sorry about the confusion, I was checking a config generated with the old clusterctl and didn't realize this only happens with the newer one.

Reply:
I didn't have an old version to check against, but it seemed like it needed to be fixed either way and shouldn't cause a backwards compatibility issue.

Reply:
Are you testing in an environment that is using a version of kustomize later than 4.0.4? Versions after that have a bug that don't do the line wrapping well. That's why our make script is downloading 4.0.4 and not anything newer.

Reply:
I think I get the cause for this now, clusterctl is using a newer kustomize (or library associated with kustomize) inside of it that has issues with line wrapping long yaml lines. This is the same reason we use 4.0.4 and not anything newer in the Makefile. The upstream root cause is still an open issue sadly. :(

Reply:
On the plus side, with these changes we can just use the latest customize now in the makefile...I'll have to try that.

Reply:
I was using Kustomize v4.0.5 so that would explain it. I just tested with the latest Kustomize v4.5.2 and the line wrapping issue is still present, but my workaround still works so that should indeed allow for using the latest Kustomize going forward. This is probably also a good way to prevent issues as users might have newer versions of Kustomize installed.